client/app/visualizations/chart/chart-editor.html

```diff
@@ -13,7 +13,7 @@
       <a ng-click="changeTab('series')">Series</a>
     </li>
   </ul>
-  <div ng-show="currentTab == 'general'">
+  <div ng-if="currentTab == 'general'" class="m-t-10 m-b-10">
```

Comment:
Any special reason to switch to `ng-if`? I think that because those are tabs, it will be simpler/quicker to have `ng-show`, no?

Reply:
It's a king of holywar in AngularJS world - `ng-show` vs. `ng-if`. Generic recommendation is to use `ng-show` for more or less static fragments that should frequently be shown/hidden. `ng-if` works better if sub-view has a lot of bindings - because when `ng-if` condition becomes false, `ng-if` removed subtree from DOM and detaches all event listeners, watchers, destroys directives, etc. This also means that `ng-if` is bad if its condition changes too frequently. But in case of tabs, IMHO, `ng-if` is better solution - because it matches all mentioned pros - condition changes rarely, subtree contains a log of angular bingins.

Reply:
I guess it depends on what's defined as frequent... but ok, let's use ng-if.

Reply:
I think, *frequent* should be measured with comparing time of rendering subtree with interval of changing condition. In case with tabs, subtree will render, say, in 50-100ms, but user switches tabs once in, say, 5 or more seconds (I think this assumptions are more or less correct). So, comparing that values, we can see that subtree renders fast enough, but it is shown or hidden very rarely - because user will spend tens of seconds and more while editing options on tab.

Reply:
👍